TimberAPI/SpecificationSystem/Buildings/BuildingDeserializer.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using Timberborn.Persistence;
+
+namespace TimberbornAPI.SpecificationSystem.Buildings
+{
+    public class BuildingDeserializer : IObjectSerializer<Building>
+    {
+        private readonly PropertyKey<int> _scienceCostKey = new PropertyKey<int>("ScienceCost");
+        private readonly ListKey<BuildingCost> _buildingCostKey = new ListKey<BuildingCost>("BuildingCost");
+
+        private readonly BuildingCostDeserializer _buildingCostDeserializer;
+
+        public BuildingDeserializer(BuildingCostDeserializer buildingCostDeserializer)
+        {
+            _buildingCostDeserializer = buildingCostDeserializer;
+        }
+
+        /// <summary>
+        /// This class only deserializes specification jsons, so this is not used
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="objectSaver"></param>
+        public void Serialize(Building value, IObjectSaver objectSaver)
+        {
+            throw new NotSupportedException();
+        }
+
+        public Obsoletable<Building> Deserialize(IObjectLoader objectLoader)
+        {
+            var scienceCost = objectLoader.Has(_scienceCostKey)
```

Comment:
Not sure if every Deserializer needs to have these, but I had some problems when implementing, that if a Building didn't have some combonent that was used to create BuildingSpecs, it would throw an error. These fallbacks were made to ensure that feature wouldn't crash